include/boost/geometry/algorithms/detail/calculate_sum.hpp

```diff
@@ -30,7 +34,7 @@ class calculate_polygon_sum
     template <typename ReturnType, typename Policy, typename Rings, typename Strategy>
     static inline ReturnType sum_interior_rings(Rings const& rings, Strategy const& strategy)
     {
-        ReturnType sum = ReturnType();
+        ReturnType sum = ReturnType(0);
```

Comment:
Oh, it was actually OK as it was. You may initialize a template parameter using ReturnType(). It gets the zero automatically. That is made for templates. For complex types (such as high precision), it is often more convenient and does not need a conversion from 0 to the type (or it might even be impossible).